server/ServerFSM.cpp

```diff
@@ -748,6 +748,14 @@ void MPLobby::ValidateClientLimits() {
             ai_count++;
     }
 
+    const int human_connected_count = human_count;
+
+    // for load game consider as human all non-AI empires
+    // because human player could connect later in game
```

Comment:
what is the consequence of counting human players this way...?

Reply:
This allows to check restrictions on human player count. As an example if it capped by 5 but 4 human players loads game with 10 human empires the game will exceed limit without this check when all players connect to the game.

Reply:
I thought the idea was that a multi-player save could be loaded with just a single player connected, who could play their turn then disconnect, and then another player could join and play their turn, until all the turns are played, but with only a single player ever connected. Why shouldn't this be possible, even if there is a limit (of more than one) for how many players can be connected at once?

english tip: don't write "allow to check something", but rather "allow something to be checked"

Reply:
I suppose restriction works on playing game should be checked before playing game as well.
> I thought the idea was that a multi-player save could be loaded with just a single player connected, who could play their turn then disconnect, and then another player could join and play their turn, until all the turns are played, but with only a single player ever connected. 

Actually it won't work before last step of #2243 would be implemented. If server set up to stop on no players connected then next load resets ready status of empires.

> Why shouldn't this be possible, even if there is a limit (of more than one) for how many players can be connected at once?

Because the server's owner sets up this limit.